File: returnLongestNonRepeatSubString/c#/hello.cs
 string returnLongestNonRepeatSubString(string s)
        {
            if (string.IsNullOrEmpty(s))
            {
                return "";
            }

            List<string> subCollection = new List<string>();
            string currentSub = "";
            for (int i = 0; i < s.Length; i++)
            {
                char c = s[i];
                if (currentSub.IndexOf(c) == -1)
                {
                    currentSub += c;
                }
                else
                {
                    subCollection.Add(currentSub);
                    currentSub = "" + c;
                }
            }

            if (currentSub.Length > 0)
            {
                subCollection.Add(currentSub);
            }

            string longest = "";
            for (int i = 0; i < subCollection.Count; i++)
            {
                string sub = subCollection[i];
                if (sub.Length > longest.Length)
                {
                    longest = sub;
                }
            }

            return longest;
        }

File: returnLongestNonRepeatSubString/kotlin/hello.kt
fun returnLongestNonRepeatSubString(s: String?): String {
   if (s == null || s.length == 0) {
       return ""
   }

   val subCollection = ArrayList<String>()
   var currentSub = ""
   for (i in 0 until s.length) {
       val c = s[i]
       if (currentSub.indexOf(c) == -1) {
           currentSub += c
       } else {
           subCollection.add(currentSub)
           currentSub = "" + c
       }
   }

   if (currentSub.length > 0) {
       subCollection.add(currentSub)
   }

   var longest = ""
   for (i in subCollection.indices) {
       val sub = subCollection[i]
       if (sub.length > longest.length) {
           longest = sub
       }
   }

   return longest
}
